### DISCUSSION TITLE
Can explosion regeneration happen inside of Towns?

### DISCUSSION THREAD

**Discussion by sexibatka** [TownyAdvanced/Towny#4693] [Q&A] [answered]
is it possible to keep explosions on in a town and have creeper explosions regenerate in towns? they regenerate in the wilderness but they don't do in towns

**Comment by LlmDl** [comment, marked as answer]
The explosion regen only operates inside of Towns during EventWar. It is not possible at this time to have them regenerate in Towns outside of EventWar.

**Reply by sexibatka** [reply]
ok do you have any plans to make it so?

**Reply by LlmDl** [reply]
There has to be a review of the system now that BlockData is a thing. EventWar currently doesn't regenerate certain blocks (like doors, chests and other blocks,) because they did not regenerate well. Now that blocks have BlockData it might be more reliable. 

We wouldn't want to have a system that unreliably regenerates things inside of Towns. There's also the issue of some blocks which require a surface (like pressure plates,) losing their surface and breaking but not counting towards the regeneration. This would mean that some blocks in towns would not regenerate, and it is why this is not a high-priority feature being considered.

**Reply by sexibatka** [reply]
ok thank you
